src/player_activity.cpp

```diff
@@ -33,6 +33,37 @@ player_activity::player_activity( activity_type t, int turns, int Index, int pos
 {
 }
 
+player_activity::player_activity( const player_activity &rhs )
+    : type( rhs.type ), moves_total( rhs.moves_total ), moves_left( rhs.moves_left ),
+      index( rhs.index ), position( rhs.position ), name( rhs.name ),
+      ignore_trivial( rhs.ignore_trivial ), values( rhs.values ), str_values( rhs.str_values ),
+      coords( rhs.coords ), placement( rhs.placement ),
+      warned_of_proximity( rhs.warned_of_proximity ), auto_resume( rhs.auto_resume )
+{
+    target = rhs.target.clone();
```

Comment:
> Could probably be moved to the initializer list as target( rhs.target.clone() ).

`target` is probably going to become `std::vector<item_location>`

> We could also drop the whole no copying constructor thing from item locations

Possibly but it greatly limits the risk of accidentally storing a stale `item_location`. I agree that's entirely possible when using bare `item *` but any extra safety is a good thing. It forces the caller to consider ownership rather than passing it around blindly.